### DISCUSSION TITLE
Cant Verify contract on Etherscan

### DISCUSSION THREAD

**Discussion by jmanywhere** [Cyfrin/foundry-full-course-cu#93] [Q&A] [answered]
So last night I was making a custom NFT contract which I successfully deployed. however the verification failed
Console Response:
```
Submitting verification for [src/NebuloidsNFT.sol:NebuloidsNFT] "0xC077519505698c5b2b4E8171869A66402a5a3895".
Submitted contract for verification:
        Response: `OK`
        GUID: `hav4h9qfdaukumgkswex1g9hgjz4hzv6xkzbtjc3jf4lmwruga`
        URL:
        https://etherscan.io/address/0xc077519505698c5b2b4e8171869a66402a5a3895
Contract verification status:
Response: `NOTOK`
Details: `Pending in queue`
Contract verification status:
Response: `NOTOK`
Details: `Fail - Unable to verify. Please check if the correct constructor argument was entered.`
```

The command I ran in the console was:
`forge script script/DeployNebuloid.s.sol --rpc-url https://eth.public-rpc.com --private-key $Uploader --etherscan-api-key $ETHERSCAN_API_KEY --broadcast --verify`

The script: (I was reusing the default template script and was too lazy to change the names... sorry
```
// SPDX-License-Identifier: UNLICENSED
pragma solidity ^0.8.13;

import {Script} from "forge-std/Script.sol";
import {NebuloidsNFT} from "../src/NebuloidsNFT.sol";

contract CounterScript is Script {
    function run() public {
        vm.startBroadcast();
        new NebuloidsNFT(
            "ipfs://bafkreihxcamsslcsfjp2hvn42x6xh4i2jogrxjmvkx725y3nzwsfzeajfe/"
        );
        vm.stopBroadcast();
    }
}
```

after that failed, got the exact same response again after doing:
```
forge verify-contract \
    --chain-id 1 \
    --num-of-optimizations 200 \
    --watch \
    --constructor-args $(cast abi-encode "constructor(string)" "ipfs://bafkreihxcamsslcsfjp2hvn42x6xh4i2jogrxjmvkx725y3nzwsfzeajfe/") \
    --etherscan-api-key $ETHERSCAN_API_KEY \
    --compiler-version v0.8.20+commit.a1b79de6 \
    0xc077519505698c5b2b4e8171869a66402a5a3895 \
    src/NebuloidsNFT.sol:NebuloidsNFT
```

I read that foundry and etherscan divert in how imports are handled. Which in the contract itself looks like:
```
import "@solmate/tokens/ERC721.sol";
import "@solmate/auth/Owned.sol";
import "@solmate/utils/LibString.sol";
```
Any help is greatly appreciated as I would like to NOT spend gas just to redeploy via Remix, only to verify this thing.

**Comment by alymurtazamemon** [comment, marked as answer]
@jmanywhere Please run the `foundryup` command someone had the same issue and he resolve it by doing this.

You should also check/replace your etherscan API key.

And try to run this command again;
```bash
forge verify-contract --chain-id 1 --constructor-args $(cast abi-encode "constructor(string)" "ipfs://bafkreihxcamsslcsfjp2hvn42x6xh4i2jogrxjmvkx725y3nzwsfzeajfe/") --etherscan-api-key <your_etherscan_api_key> 0xc077519505698c5b2b4e8171869a66402a5a3895 src/NebuloidsNFT.sol:NebuloidsNFT

```

**Reply by jmanywhere** [reply]
This worked! Thanks!

**Reply by jmanywhere** [reply]
oh and BTW, didn't have to swap my etherscan API key.

**Reply by MattPereira** [reply]
`foundryup` command fixed auto verification for me as well with deploying the lottery contract from lesson 9 

Much appreciated @alymurtazamemon :beers:

**Reply by 0xTagger** [reply]
hey @alymurtazamemon I ran the foundryup command as well and it solved the issue of my contract not verifying for the Raffle during Lesson 9. I did not have to swap my etherscan API key or anything else.

Can you please explain why it worked only after running the foundryup command before make deploy ARGS="--network sepolia"? What is the reason I needed to run foundryup so I know for the future? Thanks!
